packages/grafana-data/src/dataframe/StreamingDataFrame.ts

```diff
@@ -42,7 +42,7 @@ function circPush(data: number[][], newData: number[][], maxLength = Infinity, d
     sliceIdx = nlen - maxLength;
   }
 
-  if (maxDelta !== Infinity) {
+  if (maxDelta !== Infinity && deltaIdx >= 0) {
```

Comment:
I keep getting burned by this crashing when -1 (no time field found yet)

Reply:
👍